### DISCUSSION TITLE
SU2-feature_adap runnig error

### DISCUSSION THREAD

**Discussion by MSLEE-AERO** [su2code/SU2#2035] [Q&A]
Hello, I'm MINSUL.

I installed SU2-feature_adap branch.

I tried to solve ONERA_M6 wing with adap_ONERAM6.cfg file.

but the problem is that whenever first iteration ends (before entering pyamg process in amg.py) ,  It shows error messages as below.

I think SU2_CFD solver shows an error when it reached the end.

The problem is that mesh_adaptation.py script also ended with returning the error message: SU2 process returned error '139

----------------------------- Solver Exit -------------------------------

Maximum number of iterations reached (ITER = 50) before convergence.
+-----------------------------------------------------------------------+
|      Convergence Field     |     Value    |   Criterion  |  Converged |
+-----------------------------------------------------------------------+
|                  Cauchy[CD]|     0.0262824|       < 1e-10|          No|
+-----------------------------------------------------------------------+
-------------------------------------------------------------------------
+-----------------------------------------------------------------------+
|        File Writing Summary       |              Filename             |
+-----------------------------------------------------------------------+
|SU2 restart                        |restart_flow.dat                   |
|Paraview binary                    |flow.vtk                           |
|Paraview binary surface            |surface_flow.vtk                   |
+-----------------------------------------------------------------------+

------------------------- Solver Postprocessing -------------------------
Deleted CNumerics container.
Deleted CIntegration container.

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 16845 RUNNING AT nodemaster01
=   EXIT CODE: 139
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Segmentation fault (signal 11)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions

(In the command window,)
SU2 process returned error '139'

Is there anyone who knows how to solve this problem?

Thank you.

**Comment by MSLEE-AERO** [comment]
I found out that 

in SU2_CFD/src/drivers/CDriver.cpp,
#line 415 : Solver_Postprocessing,
I think that c++ delete[] portion caused the trouble.

So I commented out those portions.

And finally, I succeed to run this code!

**Reply by bigfooted** [reply]
Hi @MSLEE-AERO, great that you got this running! I'd be really interested in seeing some results. Did you use feature_adap or feature_adap_dev? 
I think @bmunguia is still working on this in the _dev branch which is more up to date, I hope that he is still around to give us an update as well.

**Reply by MSLEE-AERO** [reply]
Hello, @bigfooted , I used SU2-feature_adap branch, and I solved the ONERA M6 wing.
I think there are a few problems in surface adaptation.

![ONERA_M6_wing_adap](https://github.com/su2code/SU2/assets/52307266/1404e342-9fcc-4d31-8ebd-58dd647393ae)
![ONERAM6_sym](https://github.com/su2code/SU2/assets/52307266/40bcfcef-7556-4cfa-8439-776f7d038b94)

So I tried to SU2-feat_adap_dev branch, but there is some allocation error, I think.
I now trying to solve that problem.

Thank you!

**Comment by casfra96** [comment]
> Hello, @bigfooted , I used SU2-feature_adap branch, and I solved the ONERA M6 wing. I think there are a few problems in surface adaptation.
> 
> 
> So I tried to SU2-feat_adap_dev branch, but there is some allocation error, I think. I now trying to solve that problem.
> 
> Thank you!

Hi, did you find a way to use SU2-feat_adap_dev branch? I get an error after the CFD initializzation (during the first mesh adaption).

Alternatively, have you managed to do a good surface refinement using SU2-feature_adap branch?
